File: Diagram/CustomDraggingService/CustomDraggingServiceVB/CustomDraggingService.vb
Imports Telerik.Windows.Diagrams.Core

#Region "CustomDraggingService"
Public Class CustomDraggingService
    Inherits DraggingService

    Private startDragPoint As Telerik.Windows.Diagrams.Core.Point

    Private _dragMode As DragMode = DragMode.Both

    Public Sub New(ByVal graph As IGraphInternal)
        MyBase.New(graph)
    End Sub

    Public Property DragMode As DragMode
        Get
            Return Me._dragMode
        End Get

        Set(ByVal value As DragMode)
            Me._dragMode = value
        End Set
    End Property

    Public Overrides Sub InitializeDrag(ByVal point As Telerik.Windows.Diagrams.Core.Point)
        Me.startDragPoint = point
        MyBase.InitializeDrag(point)
    End Sub

    Public Overrides Sub Drag(ByVal newPoint As Telerik.Windows.Diagrams.Core.Point)
        Dim dragPoint As Telerik.Windows.Diagrams.Core.Point = newPoint
        If Me.dragMode = dragMode.Horizontal Then
            dragPoint = New Telerik.Windows.Diagrams.Core.Point(newPoint.X, Me.startDragPoint.Y)
        ElseIf Me.dragMode = dragMode.Vertical Then
            dragPoint = New Telerik.Windows.Diagrams.Core.Point(Me.startDragPoint.X, newPoint.Y)
        End If

        MyBase.Drag(dragPoint)
    End Sub
End Class

Public Enum DragMode
    Both
    Horizontal
    Vertical
End Enum
#End Region

File: Diagram/CustomDraggingService/CustomDraggingService/CustomDraggingService.cs
using Telerik.Windows.Diagrams.Core;

namespace CustomDraggingService
{
    #region CustomDraggingService
    public class CustomDraggingService : DraggingService
    {
        private Telerik.Windows.Diagrams.Core.Point startDragPoint;
        private DragMode dragMode = DragMode.Both;

        public CustomDraggingService(IGraphInternal graph)
            : base(graph)
        { }

        public DragMode DragMode
        {
            get
            {
                return this.dragMode;
            }
            set
            {
                this.dragMode = value;
            }
        }

        public override void InitializeDrag(Telerik.Windows.Diagrams.Core.Point point)
        {
            this.startDragPoint = point;

            base.InitializeDrag(point);
        }

        public override void Drag(Telerik.Windows.Diagrams.Core.Point newPoint)
        {
            Telerik.Windows.Diagrams.Core.Point dragPoint = newPoint;
            if (this.DragMode == DragMode.Horizontal)
            {
                dragPoint = new Telerik.Windows.Diagrams.Core.Point(newPoint.X, this.startDragPoint.Y);
            }
            else if (this.DragMode == DragMode.Vertical)
            {
                dragPoint = new Telerik.Windows.Diagrams.Core.Point(this.startDragPoint.X, newPoint.Y);
            }

            base.Drag(dragPoint);
        }
    }

    #endregion

    #region DragModeEnum
    public enum DragMode
    {
        Both,
        Horizontal,
        Vertical,
    }
    #endregion
}
